modules/basics/src/main/java/com/opengamma/strata/basics/date/StandardDateSequences.java

```diff
@@ -64,6 +64,114 @@ public LocalDate dateMatching(YearMonth yearMonth) {
     }
   },
 
+  // IMM in Mar/Jun/Sep/Dec plus first 4 non quarterly serial months
+  QUARTERLY_IMM_4_SERIAL("Quarterly-IMM-4-Serial") {
+    @Override
+    public DateSequence baseSequence() {
+      return QUARTERLY_IMM;
+    }
+
+    @Override
+    public LocalDate next(LocalDate date) {
+      return nth(date, 1);
+    }
+
+    @Override
+    public LocalDate nextOrSame(LocalDate date) {
+      return nthOrSame(date, 1);
+    }
+
+    @Override
+    public LocalDate nth(LocalDate date, int sequenceNumber) {
+      ArgChecker.notNegativeOrZero(sequenceNumber, "sequenceNumber");
+      LocalDate baseImm = date.with(THIRD_WEDNESDAY);
+      if (!baseImm.isAfter(date)) {
+        baseImm = baseImm.plusMonths(1);
+      }
+      return shift(baseImm, sequenceNumber);
+    }
+
+    @Override
+    public LocalDate nthOrSame(LocalDate date, int sequenceNumber) {
+      ArgChecker.notNegativeOrZero(sequenceNumber, "sequenceNumber");
+      LocalDate baseImm = date.with(THIRD_WEDNESDAY);
+      if (baseImm.isBefore(date)) {
+        baseImm = baseImm.plusMonths(1);
+      }
+      return shift(baseImm, sequenceNumber);
+    }
+
+    private LocalDate shift(LocalDate base, int sequenceNumber) {
+      // first 4 serial can be expanded to first 6 serial by subsuming first two quarterly
+      if (sequenceNumber <= 6) {
+        return base.plusMonths(sequenceNumber - 1).with(THIRD_WEDNESDAY);
+      }
+      int effectiveQuarterlySequenceNumber = sequenceNumber - 4;
+      int month = base.getMonthValue();
+      int offset = (month % 3 == 0 ? 0 : 3 - month % 3) + (effectiveQuarterlySequenceNumber - 1) * 3;
```

Comment:
Finding it hard to read that first ternary operation with operator precedence, might be easier to read in separate statements?
That first part is the offset to the next quarterly, and the second is the remaining quarterlies?